table/tables/index.go

```diff
@@ -204,26 +204,44 @@ func (c *index) Create(sctx sessionctx.Context, rm kv.RetrieverMutator, indexedV
 	if err != nil {
 		return 0, err
 	}
+
+	ctx := opt.Ctx
+	if opt.Untouched {
+		// If the index kv was untouched(unchanged), and the key/value already exists in mem-buffer,
+		// no need to re-write the index kv.
+		_, err := rm.GetFromTxnMem(ctx, key)
```

Comment:
re-write the index kv doesn't cost a lot.

Reply:
This is for the below test, not for optimization.
```sql
	// Test insert and update with untouched index.
	tk.MustExec("drop table if exists t1")
	tk.MustExec("create table t1 (a int,b int,c int,index idx(b));")
	tk.MustExec("begin;")
	tk.MustExec("insert into t1 values (1, 1, 1), (2, 2, 2);")
	tk.MustExec("update t1 set c=c+1 where a=1;")
	tk.MustQuery("select * from t1 use index(idx);").Check(testkit.Rows("1 1 2", "2 2 2"))
	tk.MustExec("commit")
	tk.MustExec("admin check table t1;")
```